acra-censor/handlers/blacklist_handler.go

```diff
@@ -160,153 +157,208 @@ func (handler *BlacklistHandler) handleParenTables(statement *sqlparser.ParenTab
-	parsedSelectQuery, err := sqlparser.Parse(selectQuery)
-	if err != nil {
-		log.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorQueryParseError).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorQueryParseError).WithError(err).Errorln("Can't parse query in blacklist handler to check is it dangerous select")
-		return true, ErrQuerySyntaxError
-	}
-	evaluatedStmt := parsedSelectQuery.(*sqlparser.Select)
-	if evaluatedStmt.Where != nil {
-		if strings.EqualFold(sqlparser.String(forbiddenWhere), sqlparser.String(evaluatedStmt.Where.Expr)) {
-			if handler.isForbiddenTableAccess(evaluatedStmt.From, forbiddenTables) {
-				if handler.isForbiddenColumnAccess(evaluatedStmt.SelectExprs, forbiddenColumns) {
-					return true, nil
-				}
-			}
+//handle %%SELECT%% pattern
+func handleSelectPattern(queryNodes, patternNodes []sqlparser.SQLNode) (patternDetected bool, matchOccurred bool) {
+	var emptySelect *sqlparser.Select
+	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(emptySelect) && reflect.TypeOf(patternNodes[0]) == reflect.TypeOf(emptySelect) {
+		if strings.EqualFold(sqlparser.String(patternNodes[0].(*sqlparser.Select).SelectExprs), SelectConfigPlaceholderReplacerPart2) {
+			return true, true
+		} else {
+			return true, false
 		}
-	} else {
-		if handler.isForbiddenTableAccess(evaluatedStmt.From, forbiddenTables) {
-			if handler.isForbiddenColumnAccess(evaluatedStmt.SelectExprs, forbiddenColumns) {
-				return true, nil
+	}
+	return false, false
+}
+
+//handle SELECT %%COLUMN%% .. %%COLUMN%% pattern
+func handleSelectColumnPattern(queryNodes, patternNodes []sqlparser.SQLNode) (patternDetected bool, matchOccurred bool) {
+	var emptySelect *sqlparser.Select
+	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(emptySelect) && reflect.TypeOf(patternNodes[0]) == reflect.TypeOf(emptySelect) {
+		if strings.Contains(sqlparser.String(patternNodes[0].(*sqlparser.Select).SelectExprs), ColumnConfigPlaceholderReplacer) {
+			if len(patternNodes[0].(*sqlparser.Select).SelectExprs) == len(queryNodes[0].(*sqlparser.Select).SelectExprs) {
+				return true, true
+			} else {
+				return true, false
 			}
 		}
 	}
-	return false, nil
+	return false, false
 }
 
-func (handler *BlacklistHandler) isForbiddenTableAccess(tablesToEvaluate sqlparser.TableExprs, forbiddenTables sqlparser.TableExprs) bool {
-	for _, tableToEvaluate := range tablesToEvaluate {
-		for _, forbiddenTable := range forbiddenTables {
-			if reflect.DeepEqual(tableToEvaluate.(*sqlparser.AliasedTableExpr).Expr, forbiddenTable.(*sqlparser.AliasedTableExpr).Expr) {
-				return true
+//handle SELECT a, b from t %%WHERE%% pattern
+func handleSelectWherePattern(queryNodes, patternNodes []sqlparser.SQLNode) (patternDetected bool, matchOccurred bool) {
+	var emptySelect *sqlparser.Select
+	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(emptySelect) && reflect.TypeOf(patternNodes[0]) == reflect.TypeOf(emptySelect) {
```

Comment:
we use this check > 1 times. what about to add some helper like `func isSelectQuery(node sqlparser.SQLNode)` and then use:
`if isSelectQuery(queryNodes[0]) && isSelectQuery(patternNodes[0])` ? it's more readable and we encapsulate logic of detecting query type